metals/src/main/scala/scala/meta/internal/metals/MetalsServerConfig.scala

```diff
@@ -55,6 +57,14 @@ final case class MetalsServerConfig(
       "metals.warnings",
       default = true
     ),
+    openFilesOnRenames: Boolean = MetalsServerConfig.binaryOption(
+      "metals.open-files-on-rename",
+      default = false
+    ),
+    renameFileThreshold: Int = MetalsServerConfig.intOption(
+      "metals.rename-file-threshold",
+      default = 300
```

Comment:
300 is around the moment that VS Code is not yet totally killed and the previews still make sense. Around 1000 everything went bonkers.